subprojects/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ToolingApiDistributionResolver.groovy

```diff
@@ -49,15 +50,23 @@ class ToolingApiDistributionResolver {
         if (!distributions[toolingApiVersion]) {
             if (useToolingApiFromTestClasspath(toolingApiVersion)) {
                 distributions[toolingApiVersion] = new TestClasspathToolingApiDistribution()
+            } else if (LocallyBuiltGradleDistribution.isLocallyBuiltVersion(toolingApiVersion)) {
+                File toolingApiJar = LocallyBuiltGradleDistribution.getToolingApiJar(toolingApiVersion)
+                List<File> slf4j = resolveDependency("org.slf4j:slf4j-api:1.7.25").toList()
```

Comment:
Why is this change necessary?  I don't think we should be mixing in dependencies like this. 

It looks like everything else is coming in via the classpath or is packaged in the tooling API jar?

Reply:
What do you mean with "everything else is coming in via the classpath or is packaged in the tooling API jar"? What we are doing here is not a change in existing behaviour, it is something new.

We had:
- tooling api from classpath
- tooling api from maven repository which depends on `slf4j-api` - note that the fat jar does not include slf4j.

Now we additionally have: locally built shaded tooling api jar without the pom metadata. In order for this to work, we need to add an `slf4j-api` jar to its classpath, right?

Should we register a new `Upload` task in `tooling-api.gradle.kts` instead which uploads the tooling api jar with metadata to a local file repository at a given location? Then we could consume the tooling api jar from this local file repository.

What do you suggest?

Reply:
I think trying to keep this as simple as we can (from the classpath or resolved from a repository) sounds best.  Otherwise, this is another place we need to update dependencies (or know that dependencies could come from).